js/viz/gauges/utils/resolve_label_overlapping.js

```diff
@@ -0,0 +1,115 @@
+const ARC_COORD_PREC = 5;
+
+import { normalizeAngle, normalizeArcParams } from '../../core/utils';
+
+export function dividePoints(bars) {
+    return bars.reduce(function(stackBars, bar) {
+        const angle = normalizeAngle(bar._angle);
```

Comment:
it's not ok that we use a private api in utils.
We should implement public getters and tests it to make the code more safe and ensure we understand the code's responsibility zone.
But if it's by design for chart now - ok

Reply:
Explain the moment of the problem, as I understand you, is that unclean functions are used that change the initial state of the object and because they have a separate export it becomes unsafe?

Reply:
Smth like this, yes. Utils should be a pure functions - they gets some data, calculates smth and returns a result. They shouldn't get another class instance as a prop and mutate it. If some function need access to other class' members this function should mostly be a method of this class :) We should not mutate instance's private members outside of the class implementation.

Please, call me if u need further assistance. U think it'll be faster to discuss it in voice

Reply:
Finished the discussion at the meeting